src/Models/ArduinoDeviceBase.ts

```diff
@@ -160,20 +160,46 @@ export abstract class ArduinoDeviceBase implements Device {
   abstract get version(): string;
 
   private async writeCppPropertiesFile(
-      rootPath: string, boardId: string, type: ScaffoldType,
-      cppPropertiesTemplateFileName: string,
-      cppPropertiesFilePath: string): Promise<void> {
+      boardId: string, type: ScaffoldType, plat: string): Promise<void> {
+    const cppPropertiesFilePath =
+        path.join(this.vscodeFolderPath, constants.cppPropertiesFileName);
+
+    if (await FileUtility.directoryExists(type, cppPropertiesFilePath)) {
+      return;
+    }
+
+    let cppPropertiesTemplateFileName: string;
+    let changeRootPath = false;
+    let rootPath: string = await IoTWorkbenchSettings.getOs();
+    if (plat === OSPlatform.WIN32) {
+      rootPath = path.join(rootPath, 'AppData', 'Local').replace(/\\/g, '\\\\');
+      cppPropertiesTemplateFileName = constants.cppPropertiesFileNameWin;
+      changeRootPath = true;
+    } else if (plat === OSPlatform.LINUX) {
+      cppPropertiesTemplateFileName = constants.cppPropertiesFileNameLinux;
+      changeRootPath = true;
+    } else {
+      // TODO: Let's use the MacOS template file for OS that is not win32/linux.
+      // Revisit this part if want to support other OS.
+      cppPropertiesTemplateFileName = constants.cppPropertiesFileNameMac;
+    }
+
     const cppPropertiesTemplateFilePath =
         this.extensionContext.asAbsolutePath(path.join(
             FileNames.resourcesFolderName, FileNames.templatesFolderName,
             boardId, cppPropertiesTemplateFileName));
     const propertiesContent =
-        fs.readFileSync(cppPropertiesTemplateFilePath).toString();
+        await FileUtility.readFile(type, cppPropertiesTemplateFilePath);
+    const propertiesContentString = propertiesContent.toString();

```

Comment:
I have seen many code logic as this to replace version value in file in the past few reviews, can we make it a common function if we frequently use it?

Reply:
Yes. This is **the** common function.

Reply:
Yes. This is the common function.

Reply:
> I have seen many code logic as this to replace version value in file in the past few reviews, can we make it a common function if we frequently use it?

Yes. This is the common function.